pkg/mimirtool/client/backfill.go

```diff
@@ -100,11 +104,25 @@ func (c *MimirClient) backfillBlock(ctx context.Context, blockDir string, logctx
 		}
 	}
 
-	resp, err = c.doRequest(ctx, path.Join(endpointPrefix, url.PathEscape(blockID), finishBlockUpload), http.MethodPost, nil, -1)
-	if err != nil {
-		return errors.Wrap(err, "request to finish block upload failed")
+	retryConfig := backoff.Config{
+		MinBackoff: 5 * time.Second,
+		MaxBackoff: 1 * time.Minute,
+		MaxRetries: maxRetries,
```

Comment:
If the sleep time provided by the user should be respected then the backoff can be removed in favor of the direct sleep, apologies if I confused the issue by suggesting it. The benefit to using backoff would be respecting context cancellation and randomly distributing the checks to find when validation slots open since there is no correct choice for sleep time.

Reply:
I'm going to go back to a regular sleep to be consistent with the CLI flag and what we do while waiting for validation to finish.